File: T2/Etapa 2/src/main/java/com/example/t2/VideoPublisher.java
package com.example.t2;

import javafx.scene.control.Label;
import javafx.scene.control.TextField;
import javafx.scene.layout.HBox;

public class VideoPublisher extends Publisher {
    public VideoPublisher(String name, Broker broker, String topicName) {
        super(name, broker, topicName);
        //??
        Label info = new Label("Pub[" + getName() + "] → " + topicName + ":");

        message = new TextField();
        message.setPromptText("Enter video URL and press ENTER");
        message.setOnAction(e -> {
            String url = message.getText().trim();
            if (!url.isEmpty()) {
                publishNewEvent(url);
                message.clear();
            }
        });

        view = new HBox(10, info, message);
    }
    public HBox getView(){return view;}
    private HBox view;
    private TextField message;
}

File: T3/Entrega/videopublisher.cpp
#include "videopublisher.h"
#include "broker.h"

VideoPublisher::VideoPublisher(QObject* parent)
    : QObject(parent), Publisher("VideoPublisher", Broker::getInstance().getTopic("video"))
{
}

void VideoPublisher::publishUrl(const QString& url) {
    if (!url.isEmpty()) {
        publish(url);
    }
}
